flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/NullableIfArgsTypeStrategy.java

```diff
@@ -70,12 +75,24 @@ public NullableIfArgsTypeStrategy(
                                                     .orElse(argumentDataTypes.size()),
                                             argumentDataTypes.size());
 
-                            final boolean isNullableArgument =
-                                    IntStream.range(fromArg, toArg)
-                                            .mapToObj(argumentDataTypes::get)
-                                            .anyMatch(
-                                                    dataType ->
-                                                            dataType.getLogicalType().isNullable());
+                            final boolean isNullableArgument;
+                            if (nullableIfAllArgsNullable) {
+                                isNullableArgument =
+                                        IntStream.range(fromArg, toArg)
+                                                .mapToObj(argumentDataTypes::get)
+                                                .allMatch(
```

Comment:
nit: here and below, we can have Spotify make this a bit prettier by using another `map`:

```
IntStream.range(fromArg, toArg)
    .mapToObj(argumentDataTypes::get)
    .map(DataType::getLogicalType)
    .allMatch(LogicalType::isNullable);
```